overlord/snapstate/snapstate.go

```diff
@@ -447,3 +447,39 @@ func GadgetInfo(s *state.State) (*snap.Info, error) {
 
 	return nil, state.ErrNoState
 }
+
+// MigrateToTypeInState migrates the state to have a snap type in the snap state of each setup snap.
+func MigrateToTypeInState(s *state.State, commit func()) error {
+	var stateMap map[string]*SnapState
+
+	s.Lock()
+	err := s.Get("snaps", &stateMap)
```

Comment:
@mvo5 if you have a suggestion, maybe there's a clearer way in the doc comment to make clear that this is public only because of the packages responsibility split but is meant to be used only as a migration.

maybe: "implements a state migration" ? would be better with a warning "To be used from overlord/migrations.go only" ?